rct229/rulesets/ashrae9012019/section1/section1rule2.py

```diff
@@ -0,0 +1,125 @@
+from pydash import filter_
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.ruleset_model_factory import produce_ruleset_model_instance
+from rct229.rulesets.ashrae9012019 import (
+    BASELINE_0,
+    BASELINE_90,
+    BASELINE_180,
+    BASELINE_270,
+    PROPOSED,
+    USER,
+)
+from rct229.utils.assertions import assert_
+from rct229.utils.jsonpath_utils import find_one
+from rct229.utils.std_comparisons import std_equal
+
+
+class Section1Rule2(RuleDefinitionBase):
+    """Rule 2 of ASHRAE 90.1-2019 Appendix G Section 1 (Performance Calculations)"""
+
+    def __init__(self):
+        super(Section1Rule2, self).__init__(
+            rmrs_used=produce_ruleset_model_instance(
+                USER=True,
+                BASELINE_0=True,
+                BASELINE_90=True,
+                BASELINE_180=True,
+                BASELINE_270=True,
+                PROPOSED=True,
+            ),
+            rmrs_used_optional=produce_ruleset_model_instance(
+                USER=True,
+                BASELINE_0=True,
+                BASELINE_90=True,
+                BASELINE_180=True,
+                BASELINE_270=True,
+                PROPOSED=True,
+            ),
```

Comment:
In the optional dict, you do not need to set `USER`, `BASELINE_0`, and `PROPOSED` because these are required in the models.

Reply:
Addressed.

Reply:
@yunjoonjung-PNNL 
I see the reason why set `USER` to optional. It is because the test case does not have a user RMDs.
@yunjoonjung-PNNL could you test if just keep `USER` to be optional but make `BASELINE_0` and `PROPOSED` to be mandatory, will it pass the CI?
@jugonzal07, is it possible to make the spreadsheet generate 3 RMDs? My guess is it could require a lot effort but is it possible to add a script to copy a proposed case for user case in the post processing script?

Reply:
@weilixu When I kept ```USER``` optional, and made ```BASELINE_0``` and ```PROPOSED``` mandatory, this ruleset fails.

Question) do ```USER```, ```BASELINE_0```, and ```PROPOSED``` have to always exist in this ruleset? My understanding is it's ok to have just one of ```USER```, ```BASELINE_0```, ```BASELIN_90```, ```BASELINE_180```, ```BASELINE_270```, AND ```PROPOSED``` because of [this line in the RDS](https://github.com/pnnl/ruleset-checking-tool/blob/develop/docs/section1/Rule1-2.md#:~:text=the%20various%20outputs-,(it%20only%20needs%20to%20exist%20in%20at%20least%201%20output),-%3A%20pci_set%20%3D%20set).

Reply:
Ok, this line is a bit misleading.
Because the way how schema structures the output, so the information in the output is repeated in every RMD model.
That is why it says it only needs exist in one model.

My understanding is since we are adding these into the rule definition as part of evaluation. The RMDs themselves shall exist. This will include the mandatory set (`USER`, `PROPOSED`, `BASELINE_0`). The line is saying the output data in the RMDs can be optional and only need one exist among the 3 or 6 RMDs.

Reply:
Thank you for this Weili.

I made a PR to address this discussion ([HERE](https://github.com/pnnl/ruleset-checking-tool/pull/1320)). 

I believe RMDs in other sections don't always include all the ```USER```, ```BASELINE_0```, and ```PROPOSED```. Should we include all of these in RMDs in other sections as well?

Reply:
> @yunjoonjung-PNNL I see the reason why set `USER` to optional. It is because the test case does not have a user RMDs. @yunjoonjung-PNNL could you test if just keep `USER` to be optional but make `BASELINE_0` and `PROPOSED` to be mandatory, will it pass the CI? @jugonzal07, is it possible to make the spreadsheet generate 3 RMDs? My guess is it could require a lot effort but is it possible to add a script to copy a proposed case for user case in the post processing script?

Generating `USER`, `BASELINE_0`, and `PROPOSED` is no problem, we can do that easily with the spreadsheet. We just overwrite a formula. The challenge only comes about if we need to generate different versions of `BASELINE_X`. As those stand, they appear to be duplicated when ingesting ruletest JSONs:
https://github.com/pnnl/ruleset-checking-tool/blob/develop/rct229/ruletest_engine/ruletest_rmd_factory.py#L44-L72